code/modules/mob/living/carbon/human/species_types/shadowpeople.dm

```diff
@@ -223,16 +220,29 @@
 		M.set_light(-M.lights_power)
 		if(M.occupant)
 			M.lights_action.Remove(M.occupant)
-		for(var/obj/item/O in AM.GetAllContents())
-			if(O.light_range && O.light_power)
-				disintegrate(O, M)
+		for(var/obj/item/O in M.GetAllContents())
+			light_item_check(O, M)
 
 	else if(istype(AM, /obj/machinery/light))
 		var/obj/machinery/light/L = AM
 		if(L.status == 1)
 			return
-		disintegrate(L.drop_light_tube(), AM)
+		disintegrate(L.drop_light_tube(), L)
 
+///checks if the item has an active light, and destroy the light source if it does.
+/obj/item/light_eater/proc/light_item_check(obj/item/I, atom/A)
+	if(!isitem(I))
+		return
+	if(I.light_range && I.light_power)
+		disintegrate(I, A)
+	else if(istype(I, /obj/item/gun))
+		var/obj/item/gun/G = I
+		if(G.gun_light?.on)
```

Comment:
why are you checking if on is null in an if block?

Reply:
What do you mean? They're checking if the gun has a `gun_light`, and if so, if it's on.

Reply:
Ah I see, I got twisted on the syntax